pkg/kubelet/kubelet.go

```diff
@@ -2426,28 +2433,40 @@ func (kl *Kubelet) cleanUpContainersInPod(podID types.UID, exitedContainerID str
 }
 
 // fastStatusUpdateOnce starts a loop that checks the internal node indexer cache for when a CIDR
-// is applied  and tries to update pod CIDR immediately. After pod CIDR is updated it fires off
-// a runtime update and a node status update. Function returns after one successful node status update.
+// is applied and tries to update pod CIDR immediately. Given the pod CIDR is applied, it keeps trying
+// to update pod CIDR then container runtime status and node status, until node status becomes ready.
+// Function returns after one successful node status update that gets the node to the ready status.
 // Function is executed only during Kubelet start which improves latency to ready node by updating
 // pod CIDR, runtime status and node statuses ASAP.
 func (kl *Kubelet) fastStatusUpdateOnce() {
+	defer func() {
+		kl.containerRuntimeReadyExpected = true
+	}()
+
+	start := time.Now()
+
 	for {
-		time.Sleep(100 * time.Millisecond)
+		time.Sleep(200 * time.Millisecond)
+		if time.Since(start) > nodeReadyGracePeriod {
```

Comment:
I think that this is too much, nodeReadyGracePeriod = 120 * time.Second, these are a lot of apiserver calls, there is some helper for implementin backoffs

Reply:
in a cluster with N nodes trying to update the status every 200ms after a master reboot, this can cause that the master isnot able to boot with all this load

Reply:
ok,  talked offline, GetNode() uses the Lister, is not a problem, it is `kl.syncNodeStatus()` the one that does a lot of apiserver calls 🤔

Reply:
actually, we can update the status, but the change is not reflected immediately in the cache (the change has to be propagated from the apiserver ), so it is probably will try two update twice some times

Reply:
@wojtek-t @mborsz for sig-scalability opinions

Reply:
Excessive apiserver call addressed in https://github.com/kubernetes/kubernetes/pull/113188 - it will still suffer from propagation delay though. I'm thinking about a way to inject updated node into nodeLister upon successful patch but couldn't find a way (but should we do this?).

Reply:
@marseel can you help here?

Reply:
> in a cluster with N nodes trying to update the status every 200ms after a master reboot, this can cause that the master isnot able to boot with all this load

A master reboot won't trigger anything. This logic only runs at node startup. In the case of nodepool scaling up this can be the problem though.